### PR TITLE
8283465: Character.UnicodeBlock.NUM_ENTITIES is out of date

### PR DESCRIPTION
**Pull request by naotoj**
Fixing the out-of-date number of entries in `Character.UnicodeBlock.NUM_ENTITIES` field. The regression test has been renamed and now repurposed just to examine whether the `NUM_ENTITIES` correctly has the `map.size()` value.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283465](https://bugs.openjdk.java.net/browse/JDK-8283465): Character.UnicodeBlock.NUM_ENTITIES is out of date

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)
 * [Stuart Marks](https://openjdk.java.net/census#smarks) (@stuart-marks - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7909/head:pull/7909` \
`$ git checkout pull/7909`

Update a local copy of the PR: \
`$ git checkout pull/7909` \
`$ git pull https://git.openjdk.java.net/jdk pull/7909/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7909`

View PR using the GUI difftool: \
`$ git pr show -t 7909`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7909.diff">https://git.openjdk.java.net/jdk/pull/7909.diff</a>

</details>

### CODE REVIEW

**Review by bplb — APPROVED** [review]

File: src/java.base/share/classes/java/lang/Character.java [hunk 1]
```diff
@@ -737,10 +737,10 @@ public final String toString() {
      */
     public static final class UnicodeBlock extends Subset {
         /**
-         * 696 - the expected number of entities
+         * 737 - the expected number of entities
```
Comment (stuart-marks) [inline src/java.base/share/classes/java/lang/Character.java:740]: Just a quibble about this comment... it's probably not worth repeating the actual value. But it probably is worth mentioning that the actual value should (or must) match the number of entries added to the map by constructors called from the static initializers in this class. Whenever aliases or new blocks are added, this number must be adjusted.

**Review by stuart-marks — APPROVED** [review]
Thanks for doing this! I was going to offer to do this myself but you beat me to it. Rewritten test looks great.

File: src/java.base/share/classes/java/lang/Character.java (hunk 1 above)
Comment (naotoj) [reply src/java.base/share/classes/java/lang/Character.java]: Not a "quibble" at all.  In fact, I thought the same just after I submitted the PR that the number in the comment would be easily overlooked and got stale, which would defy this cleanup. Removed the actual number and put some explanation there.

**Review by bplb — APPROVED** [review]
Added comments look good.

**Review by irisclark — APPROVED** [review]

File: src/java.base/share/classes/java/lang/Character.java (hunk 1 above)
Comment (stuart-marks) [reply src/java.base/share/classes/java/lang/Character.java]: Thanks, looks good.

**Review by stuart-marks — APPROVED** [review]

**Pull request closed without merging**